src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsLambda/HandlerMethodWrapper.cs

```diff
@@ -0,0 +1,332 @@
+            {
+                if (HasContext() && (ContextIdx < instrumentedMethodCall.MethodCall.MethodArguments.Length))
+                {
+                    return _requestIdGetter(instrumentedMethodCall.MethodCall.MethodArguments[ContextIdx]);
+                }
+                return null;
+            }
+
+            public object GetInputObject(InstrumentedMethodCall instrumentedMethodCall)
+            {
+                if (HasInputObject() && (InputIdx < instrumentedMethodCall.MethodCall.MethodArguments.Length))
+                {
+                    return instrumentedMethodCall.MethodCall.MethodArguments[InputIdx];
+                }
+                return null;
+            }
+        }
+
+        private List<string> _webResponseHeaders = ["content-type", "content-length"];
+
+        private static Func<object, object> _getRequestResponseFromGeneric;
+        private static object _initLock = new object();
+        private static FunctionDetails _functionDetails = null;
+
+        public bool IsTransactionRequired => false;
+
+        private static bool _coldStart = true;
+        private ConcurrentHashSet<string> _unexpectedResponseTypes = new();
+        private ConcurrentHashSet<string> _unsupportedInputTypes = new();
+
+        private static bool IsColdStart => _coldStart && !(_coldStart = false);
+
+        public CanWrapResponse CanWrap(InstrumentedMethodInfo methodInfo)
+        {
+            return new CanWrapResponse("NewRelic.Providers.Wrapper.AwsLambda.HandlerMethod".Equals(methodInfo.RequestedWrapperName));
+        }
+
+        private void InitLambdaData(InstrumentedMethodCall instrumentedMethodCall, IAgent agent)
+        {
+            _functionDetails = new FunctionDetails();
+
+            for (int idx = 0; idx < instrumentedMethodCall.MethodCall.MethodArguments.Length; idx++)
+            {
+                var arg = instrumentedMethodCall.MethodCall.MethodArguments[idx];
+
+                if (!_functionDetails.HasContext())
+                {
+                    var iLambdaContext = arg.GetType().GetInterface("ILambdaContext");
+                    if (iLambdaContext != null)
+                    {
+                        _functionDetails.SetContext(arg, idx);
+                        continue; // go to the next arg
+                    }
+                }
+
+                string name = arg.GetType().FullName;
+                agent.Logger.Log(Agent.Extensions.Logging.Level.Debug, $"Checking parameter: {name}");
+                if (_functionDetails.SetEventType(name, idx))
+                {
+                    agent.Logger.Log(Agent.Extensions.Logging.Level.Debug, $"Supported Event Type found: {_functionDetails.EventType}");
+                }
+                else if (!_unsupportedInputTypes.Contains(name))
+                {
+                    agent.Logger.Log(Agent.Extensions.Logging.Level.Warn, $"Unsupported input object type: {name}. Unable to provide additional instrumentation.");
```

Comment:
Should this be a warning? Customers using "unsupported" types will get flooded with messages.

Reply:
In the short term, I think we want to know when customers are using types we don't support, but I'd also be ok with reducing it to Debug. You'll only see it once, though -- that's what `_unsupportedInputTypes` is tracking.

Reply:
It's fine as-is for now.